### ISSUE TITLE
Libary wall enabled with full search on second up.

### ISSUE THREAD

**Issue by QuickSwitchLK** [jurialmunkey/skin.arctic.fuse#379]
### Skin section

Library

### Feature description

Hi Jurialmunkey, I have been sitting back as you have been doing some major navigation changes and improvements. I love the new full search on up with the alphabet because I use Kodi with a remote and no keyboard, but I also love a wide library view of posters when just flicking through. Would it be possible to have an option of both?

First up in library would show wall view and place you on page select as it dose now when enabled. ( See Image Below )

![Screenshot 2024-01-01 192847](https://github.com/jurialmunkey/skin.arctic.fuse/assets/34122743/f403c2fc-13d4-4473-ab14-6d847bc080f6)

Second press of up puts selection in the search box as it does now, but the full search options appear with the wall dropping down to make room. ( See Image Below )

![Screenshot 2024-01-01 193037](https://github.com/jurialmunkey/skin.arctic.fuse/assets/34122743/780903a2-d23c-404e-91da-fca0d19cb462)

Then on down from the last search option ( currently page selection ) it stays as a wall to select what was found ( Example "Harry" ). ( See Image Below )

![Screenshot 2024-01-01 194843](https://github.com/jurialmunkey/skin.arctic.fuse/assets/34122743/3953338b-2982-4563-b77b-0741ab0d94ec)

Currently it does not stay as wall and goes back to a single line for movie selection.

### Why is this feature needed? Please justify and explain.

It would be nice since I don't use a keyboard for text entry and I like wide poster view.

### Screenshots and Additional Info

Also with the search as it stands, I have to press up 5 times to get to the search box ( See Image )
Is there a reason when up is pressed from page select it does not start off in the search box?

![Screenshot 2024-01-01 193037](https://github.com/jurialmunkey/skin.arctic.fuse/assets/34122743/0ae215a5-d784-4f2a-8af3-1371c9fa73df)

### Checklist

- [X] I have made a separate issue for each feature I am requesting.
- [X] I have searched the issues and this feature has not been requested.
- [X] I am using the most recent skin version.

**Comment by jurialmunkey** [comment]
>  Is there a reason when up is pressed from page select it does not start off in the search box?

Because that's not the next control up. It'd make no sense to jump 5 rows of controls. 

Up always focuses scrollbar first, then you move up/down from there. I figure scrollbar is what gets most used so that's why it is first when rather than higher up like it with the wall.

PS It's technically a filter, not search. It will only find items that are already in the current view, not new items like search does.

> Second press of up puts selection in the search box as it does now, but the full search options appear with the wall dropping down to make room.

I did have a similar idea to this originally but it's mostly just been a case of haven't been able to figure out a way to do it smoothly yet. Wall animation is a bit complicated in terms of cancelling it out and moving it etc. so it is a bit difficult to wrangle into behaving well.

**Comment by jurialmunkey** [comment]
Actually thinking about this more, why exactly would you need it to expand down at all?

With library content you can do all the filtering directly from the filter button next to the edit control. Surely this is a better overal experience because you can see the wall properly whilst you filter and make changes:

![screenshot00345](https://github.com/jurialmunkey/skin.arctic.fuse/assets/4618719/243889d8-6ee8-4441-8afb-839b8a46b19e)

**Comment by QuickSwitchLK** [comment]
All of this makes sense, thanks for taking the time to explain it. I did not know selecting filter would show these options, never tried pressing it.

I guess I like most that there is already an alphabet showing to filler buy name for the remote when not in wall mode, but it is as simple as pressing ok in the text entry box and using the on screen keyboard. Just screen get covered up a lot.

1: Is there room for a small 2 line alphabet to fit anywhere when wall enabled?

2: In previous versions you were able to highlight a movie on the second row while in wall. Was this removed so that the down = explore worked? I don't use that option as I don't like it, feels like you get lost and off track from where your were to me personally. Is it much work if you have "Contect > OnDown = Bump" then you can access both rows of the wall with the highlight box? ( Hope that makes sense )

Thanks.

**Comment by jurialmunkey** [comment]
> In previous versions you were able to highlight a movie on the second row while in wall. Was this removed so that the down = explore worked

No, nothing to do with explore.

It was removed to simplify navigation and code:

1. Navigation made no sense and would utterly confuse new users. The new navigation is more familiar (up expands, down collapses), which means new users are less likely to get tuck or have unexpected things happen.
2. It was buggy. Kodi can't do a fixed position wall control, so it was a hack attempting to fake it. Being a hack, some functions you'd normally expect to be available were impossible (e.g. context menu or mouse control) and even basic actions like select/info/move would sometimes act unpredictably.

Basically, way too many issues where parts didn't work as users would expect and there either was no solution or only a very overengineered one (and even then it'd still behave jankily).

It was an experiement. Despite my best efforts, I couldn't get it working well enough for a beta release so I dropped it.

> Is there room for a small 2 line alphabet to fit anywhere when wall enabled?

In the main view, unlikely. Possibly might consider it inside the filter panel though (not certain if it is possible).

Another idea I had is an option to have clicking the edit control flip over to the mini keyboard. That might also work (would add what you type to the view line where it says "Titles").

**Comment by QuickSwitchLK** [comment]
Navigation stuff all makes sense. Lets face it, you are very good at what you do and if it is a mess and hard to work with as time goes by, why persist.

I know most of these fine issues of navigation i have are stubborn me and 10 years ago that's how you did it in Kodi. 

> Possibly might consider it inside the filter panel though (not certain if it is possible).

> Another idea I had is an option to have clicking the edit control flip over to the mini keyboard. That might also work (would add what you type to the view line where it says "Titles").

I like both of these options if it workable.

**Comment by jurialmunkey** [comment]
> I know most of these fine issues of navigation i have are stubborn me and 10 years ago that's how you did it in Kodi.

I really do appreciate people taking the time to discuss nav issues though even if it's down to learned behaviour. It's something I want to improve in the skin so it is always useful info.

It's finding that sweet spot to balance navigation which feels natural to a new user but doesn't also completely go against muscle memory for existing users.

> I like both of these options if it workable.

I actually ended up managing to reduce the size of the keyboard quite significantly which I think helps quite a bit. Obviously still covers some of the screen but I think this might be the best of both worlds -- allows for proper keyboard with language support but also improves visibility.

![screenshot00351](https://github.com/jurialmunkey/skin.arctic.fuse/assets/4618719/5538e5ba-137a-4831-ada0-7205e0564feb)

**Comment by QuickSwitchLK** [comment]
Very nice and clean. I like it a lot. Thanks again.

 > I really do appreciate people taking the time to discuss nav issues though even if it's down to learned behaviour. It's something I want to improve in the skin so it is always useful info.

Nice to know it is useful. I do want you to know that the coding ability that you have is awesome, and the last thing I want, is for it to look like I am nit picking on your hard work.

Cheers.

**Comment by jurialmunkey** [comment]
With the mini keyboard I've decided to go back to having scrollbar ondown to simplify the navigation. I still find the wall expanding upwards ondown a bit weird, but so was having scrollbar at top (since scrollbars are nearly always bottom/right in most apps). 

Now:

UP from view goes straight to filter bar no matter settings. 
UP again from filter goes to full search (if enabled otherwise bumps).
DOWN from filter always goes back to main view.

DOWN from view goes straight to scrollbar no matter settings
DOWN again goes from scrollbar goes to explore (if enabled otherwise bumps)
UP from scrollbar always goes back to main view.

Helps to address the original couple of issues in this thread (double up for full search).  Plus reduces confusion between searching and filtering by making them look a little more distinct.
